boards/ti/lp_em_cc2340r5/lp_em_cc2340r5_defconfig

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2024 Texas Instruments Incorporated
+# Copyright (c) 2024 BayLibre, SAS
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Ensure hex-file is not filled between lower and upper flash.
+# If not, it will grow to 3.5 GB.
+CONFIG_BUILD_NO_GAP_FILL=y
+CONFIG_BUILD_OUTPUT_HEX=y
```

Comment:
why does every board need to set those? what if a custom board doesn't?

Reply:
I'm not sure, especially for CONFIG_BUILD_NO_GAP_FILL. I'm currently checking with TI.

About CONFIG_BUILD_OUTPUT_HEX, do you mean it can be defined at user app level (for instance), if ever it is needed for some reason ?

Reply:
I made the test with:
CONFIG_BUILD_NO_GAP_FILL=**n**
CONFIG_BUILD_OUTPUT_HEX=**y**

As mentioned by the comment (in the defconfig file), the output hex file is 3.7 GB (and linking step takes much more time).

IOW, if some user set CONFIG_BUILD_OUTPUT_HEX option without knowing that this CONFIG_BUILD_NO_GAP_FILL exists, building step will be much longer and he will generate a much bigger file.

Reply:
looks like a soc level choice.